BUG: RectBivariateSpline interpolation issues with new fitpack implementation [1.18.0]

### Describe your issue.

Try the following interpolation on a 2D grid with `RectBivariateSpline`. The interpolation in the final step fails with the error: `ValueError: object too deep for desired array`. The issue dissappears when using SciPy 1.17.1, so I guess the problem lies with the new fitpack implemenation added in version 1.18.

I looked for similar issues, but they were slightly different. One was about using scalars and the other about using slices. In this example, I am not using either. 

### Reproducing Code Example

```python
from scipy.interpolate import RectBivariateSpline
import numpy as np

x = np.arange(-10, 10)
y = np.arange(-20, 20)
x_mesh, y_mesh = np.meshgrid(x, y)
array = np.sin(np.sqrt(x_mesh**2 + y_mesh**2)) / np.sqrt(x_mesh**2 + y_mesh**2)

x_interpolated = np.linspace(-10, 10, endpoint=True, num=100)
y_interpolated = np.linspace(-20, 20, endpoint=True, num=100)
x_interp_mesh, y_interp_mesh = np.meshgrid(
	x_interpolated, y_interpolated, sparse=True
)

interp = RectBivariateSpline(y, x, array, s=0)
mesh_profile = interp(y_interp_mesh, x_interp_mesh)
```

### Error message

```shell
scipy.interpolate._fitpack2.RectBivariateSpline
def __call__(self, x, y, dx=0, dy=0, grid=True):
	"""
	Evaluate the spline or its derivatives at given positions.
.
.
.
		if dx or dy:
			z, ier = _fitpack.parder(tx, ty, c, kx, ky, dx, dy, x, y)
			if not ier == 0:
				raise ValueError(f"Error code returned by parder: {ier}")
		else:
>           z, ier = _fitpack.bispev(tx, ty, c, kx, ky, x, y)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E           ValueError: object too deep for desired array
```

### SciPy/NumPy/Python version and system information

```shell
1.18.0 2.5.1 sys.version_info(major=3, minor=12, micro=3, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    cython blas ilp64: false
    detection method: pkgconfig
    found: true
    has ilp64: false
    include directory: /opt/_internal/cpython-3.12.13/lib/python3.12/site-packages/scipy_openblas32/include
    lib directory: /opt/_internal/cpython-3.12.13/lib/python3.12/site-packages/scipy_openblas32/lib
    name: scipy-openblas
    openblas configuration: OpenBLAS 0.3.31.dev DYNAMIC_ARCH NO_AFFINITY Haswell MAX_THREADS=64
    pc file directory: /project/.openblas
    version: 0.3.31.dev
  lapack:
    detection method: pkgconfig
    found: true
    has ilp64: false
    include directory: /opt/_internal/cpython-3.12.13/lib/python3.12/site-packages/scipy_openblas32/include
    lib directory: /opt/_internal/cpython-3.12.13/lib/python3.12/site-packages/scipy_openblas32/lib
    name: scipy-openblas
    openblas configuration: OpenBLAS 0.3.31.dev DYNAMIC_ARCH NO_AFFINITY Haswell MAX_THREADS=64
    pc file directory: /project/.openblas
    version: 0.3.31.dev
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 3.0.4
Compilers:
  c:
    commands: cc
    linker: ld.bfd
    name: gcc
    version: 14.2.1
  c++:
    commands: c++
    linker: ld.bfd
    name: gcc
    version: 14.2.1
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.2.5
  fortran:
    commands: gfortran
    linker: ld.bfd
    name: gcc
    version: 14.2.1
  pythran:
    include directory: ../../../tmp/build-env-z5xmlso7/lib/python3.12/site-packages/pythran
    version: 0.18.1
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /tmp/build-env-z5xmlso7/bin/python
  version: '3.12'
```

--- a/scipy/interpolate/_fitpack2.py
+++ b/scipy/interpolate/_fitpack2.py
@@ -1417,6 +1417,8 @@ class _BivariateSplineBase:
         tx, ty, c = self.tck[:3]
         kx, ky = self.degrees
         if grid:
+            x = x.ravel()
+            y = y.ravel()
             if x.size == 0 or y.size == 0:
                 return np.zeros((x.size, y.size), dtype=self.tck[2].dtype)
 

--- a/scipy/interpolate/tests/test_fitpack2.py
+++ b/scipy/interpolate/tests/test_fitpack2.py
@@ -1483,6 +1483,34 @@ class TestRectBivariateSpline:
             Interpolator(GridPosLats, nonGridPosLons)
         assert "y must be strictly increasing" in str(exc_info.value)
 
+    def test_grid_sparse_meshgrid(self):
+        # gh-25730
+        x = np.arange(-10.0, 10.0)
+        y = np.arange(-20.0, 20.0)
+        lut = RectBivariateSpline(x, y, np.hypot(x[:, None], y[None, :]), s=0)
+
+        xi = np.linspace(-10.0, 10.0, 25)
+        yi = np.linspace(-20.0, 20.0, 30)
+        xs, ys = np.meshgrid(xi, yi, sparse=True, indexing="ij")
+        assert xs.ndim == 2 and ys.ndim == 2
+
+        xp_assert_close(lut(xs, ys), lut(xi, yi))
+
+    def test_not_increasing_input_2d(self):
+        # gh-25730
+        x = np.arange(5.0)
+        y = np.arange(6.0)
+        lut = RectBivariateSpline(x, y, np.hypot(x[:, None], y[None, :]), s=0)
+
+        decreasing = np.array([3.0, 1.0, 4.0])
+        with assert_raises(ValueError) as exc_info:
+            lut(decreasing[:, None], y[None, :])
+        assert "x must be strictly increasing" in str(exc_info.value)
+
+        with assert_raises(ValueError) as exc_info:
+            lut(x[:, None], decreasing[None, :])
+        assert "y must be strictly increasing" in str(exc_info.value)
+
     def _sample_large_2d_data(self, nx, ny):
         rng = np.random.default_rng(1)
         x = np.arange(nx)

--- a/scipy/interpolate/tests/test_fitpack2.py
+++ b/scipy/interpolate/tests/test_fitpack2.py
@@ -1494,7 +1494,27 @@ class TestRectBivariateSpline:
         xs, ys = np.meshgrid(xi, yi, sparse=True, indexing="ij")
         assert xs.ndim == 2 and ys.ndim == 2
 
-        xp_assert_close(lut(xs, ys), lut(xi, yi))
+        xp_assert_close(lut(xs, ys), lut(xi, yi), atol=1e-14)
+
+    def test_grid_sparse_meshgrid_values(self):
+        # gh-25730
+        # The snippet from the issue, without its denominator so that no nans
+        # enter the fit. Reference values obtained by running it on scipy 1.16.1.
+        x = np.arange(-10, 10)
+        y = np.arange(-20, 20)
+        x_mesh, y_mesh = np.meshgrid(x, y)
+        z = np.sin(np.sqrt(x_mesh**2 + y_mesh**2))
+
+        xi = np.linspace(-10, 10, endpoint=True, num=3)
+        yi = np.linspace(-20, 20, endpoint=True, num=3)
+        xs, ys = np.meshgrid(xi, yi, sparse=True)
+
+        expected = np.array([
+            [-3.61178317e-01, 9.12945251e-01, 5.94013869e-02],
+            [-5.44021111e-01, -2.15105711e-16, 4.12118485e-01],
+            [4.97086683e-01, 1.49877210e-01, 8.23386213e-01],
+        ])
+        xp_assert_close(RectBivariateSpline(y, x, z, s=0)(ys, xs), expected, atol=1e-14)
 
     def test_not_increasing_input_2d(self):
         # gh-25730

